src/fzx/tui/term_app.cpp

```diff
@@ -62,12 +64,24 @@ void TermApp::processTTY()
       updateQuery = true;
       continue;
     }
-    if (*key == kEnter || *key == kCtrlU) {
+    if (*key == kEnter) {
+      quit(true);
+      return;
+    } else if (*key == kCtrlU) {
       mLine.clear();
       updateQuery = true;
     } else if (*key == kCtrlC) {
-      quit();
+      quit(false);
       return;
+    } else if (*key == kCtrlP) {
+      mCurpos++;
+    } else if (*key == kCtrlN) {
+      if (mCurpos > 0) {
+        mCurpos--;
+      }
+    } else if (*key == kTab) {
+      mSelection.push_back(mFzx.getResult(mCurpos).mIndex);
+      mCurpos++;
```

Comment:
Needs a bound check, will fail when there are no items matched for example
```suggestion
      if (mCurpos < mFzx.resultsSize()) {
        mSelection.push_back(mFzx.getResult(mCurpos).mIndex);
        mCurpos++;
      }
```